crates/store/re_chunk/src/chunk.rs

```diff
@@ -66,70 +66,20 @@ pub type ChunkResult<T> = Result<T, ChunkError>;
 // ---
 
 #[derive(Debug, Clone, Default, PartialEq)]
-pub struct ChunkComponents(
-    // TODO(#6576): support non-list based columns?
-    //
-    // NOTE: The extra `ComponentName` layer is needed because it is very common to want to look
-    // for anything matching a `ComponentName`, without any further tags specified.
-    pub IntMap<ComponentName, IntMap<ComponentDescriptor, ArrowListArray>>,
-);
+pub struct ChunkComponents(pub IntMap<ComponentDescriptor, ArrowListArray>);
 
 impl ChunkComponents {
-    /// Like `Self::insert`, but automatically infers the [`ComponentName`] layer.
-    #[inline]
-    pub fn insert_descriptor(
-        &mut self,
-        component_desc: ComponentDescriptor,
-        list_array: ArrowListArray,
-    ) -> Option<ArrowListArray> {
-        self.0
-            .entry(component_desc.component_name)
-            .or_default()
-            .insert(component_desc, list_array)
-    }
-
     /// Returns all list arrays for the given component name.
     ///
     /// I.e semantically equivalent to `get("MyComponent:*.*")`
     #[inline]
-    pub fn get_by_component_name<'a>(
-        &'a self,
-        component_name: &ComponentName,
-    ) -> impl Iterator<Item = &'a ArrowListArray> {
-        self.get(component_name).map_or_else(
-            || itertools::Either::Left(std::iter::empty()),
-            |per_desc| itertools::Either::Right(per_desc.values()),
-        )
-    }
-
-    #[inline]
-    pub fn get_by_descriptor(
+    pub fn get_by_component_name(
         &self,
-        component_desc: &ComponentDescriptor,
-    ) -> Option<&ArrowListArray> {
-        self.get(&component_desc.component_name)
-            .and_then(|per_desc| per_desc.get(component_desc))
-    }
-
-    #[inline]
-    pub fn get_by_descriptor_mut(
-        &mut self,
-        component_desc: &ComponentDescriptor,
-    ) -> Option<&mut ArrowListArray> {
-        self.get_mut(&component_desc.component_name)
-            .and_then(|per_desc| per_desc.get_mut(component_desc))
-    }
-
-    #[inline]
-    pub fn iter_flattened(&self) -> impl Iterator<Item = (&ComponentDescriptor, &ArrowListArray)> {
-        self.0.values().flatten()
-    }
-
-    #[inline]
-    pub fn into_iter_flattened(
-        self,
-    ) -> impl Iterator<Item = (ComponentDescriptor, ArrowListArray)> {
-        self.0.into_values().flatten()
+        component_name: ComponentName,
+    ) -> impl Iterator<Item = &ArrowListArray> {
+        self.0.iter().filter_map(move |(desc, array)| {
+            (desc.component_name == component_name).then_some(array)
+        })
```

Comment:
Whenever I see a hash map lookup change to a linear search one, I always wonder if we ever run into cases where this _actually_ becomes slower and not the other way around :)

Reply:
yeah so hard to tell :/
Lotsa flux here on the way though. We'll have to do some perf checking after the storm is over.